### DISCUSSION TITLE
Share input state not working as I'd expect it to

### DISCUSSION THREAD

**Discussion by guillaumeboehm** [fcitx/fcitx5#437] [Q&A] [answered]
Hello, I'm still a bit new to IM software so I'm not sure my issue is a bug or a misunderstanding on my end. I would like to set my input method on a per window basis, so I assumed that setting the Share input state option to "No" would allow me (for a bit of details) to set once on startup the french IM on my messaging apps, Japanese IM on one of my browser window, and the default english one for all the other windows. But it seems to actually share the input state among all my windows. Am I not understanding the option correctly or is there a problem on either my configuration or the program? I'm using Arch with bspwm.

**Comment by wengxt** [comment, marked as answer]
Disclaimer first, this design is specific to fcitx that you probably won't see on other platforms.

Fcitx's behavior on this is optimized for the case that you're frequently switching between two IM, so with the default "Ctrl+space" key, it's always switching between 1st in the list and another. Sometimes fcitx refer this as "inactive" and "active".

"Inactive IM" is always the first input method in the list. And what "Active IM" points to is a global shared value. The hotkey in global configuration like "enumerate input method" will change this global value. Share state or not will only remember "active/inactive".

You may ask what if you want to set "active IM" only for the current application (or the "input context"), the way to do that is that you go to "addon -> input method selector" then you can set a hot key to set active IM only for current input context (Use 2nd option / 4th option in this addon configuration).

If you are frequently switching between 3 different input method, you might find this addon useful. 3rd and 4th option allows you to set a list of keys that always bring you to the specific input method. for example, you can set "Ctrl + 1,2,3,4" to switch to input method 1st,2rd,3rd,4th input method.

**Reply by guillaumeboehm** [reply]
Yes this seems to be working perfectly fine! Thank you very much for the detailed answer!
